ov/id_pools_ipv4_subnets.go

```diff
@@ -0,0 +1,298 @@
+	if subnets.Total > 0 {
+		return subnets.Members[0], err
+	} else {
+		return subnet, err
+	}
+}
+
+func (c *OVClient) GetIPv4Subnets(start string, count string, filter string, sort string) (SubnetList, error) {
+	var (
+		uri     = "/rest/id-pools/ipv4/subnets/"
+		q       map[string]interface{}
+		subnets SubnetList
+	)
+	q = make(map[string]interface{})
+	if len(filter) > 0 {
+		q["filter"] = filter
+	}
+
+	if sort != "" {
+		q["sort"] = sort
+	}
+
+	if start != "" {
+		q["start"] = start
+	}
+
+	if count != "" {
+		q["count"] = count
+	}
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	// Setup query
+	if len(q) > 0 {
+		c.SetQueryString(q)
+	}
+
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return subnets, err
+	}
+
+	log.Debugf("Get All Subnets", data)
+	if err := json.Unmarshal([]byte(data), &subnets); err != nil {
+		return subnets, err
+	}
+	return subnets, nil
+}
+
+func (c *OVClient) CreateIPv4Subnet(subnet Ipv4Subnet) error {
+	var (
+		uri = "/rest/id-pools/ipv4/subnets/"
+		t   = (&Task{}).NewProfileTask(c)
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t.ResetTask()
+	log.Debugf("REST : %s \n %+v\n", uri, subnet)
+	log.Debugf("task -> %+v", t)
+	data, err := c.RestAPICall(rest.POST, uri, subnet)
+	if err != nil {
+		log.Errorf("Error submitting subnet creation request: %s", err)
+		return err
+	}
+
+	log.Debugf("Response New ipv4 Subnet %s", data)
+	if err := json.Unmarshal(data, &t); err != nil {
+		log.Errorf("Error with task un-marshal: %s", err)
+		return err
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *OVClient) AllocateIpv4Subnet(id string, subnet SubnetAllocatorList) (SubnetAllocatorList, error) {
+	var (
+		uri             = "/rest/id-pools/ipv4/subnets/" + id + "/allocator"
+		t               *Task
+		subnetAllocator SubnetAllocatorList
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t = t.NewProfileTask(c)
+	t.ResetTask()
+
+	log.Debugf("REST : %s \n %+v\n", uri, subnet)
+	log.Debugf("task -> %+v", t)
+	data, err := c.RestAPICall(rest.PUT, uri, subnet)
+	if err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error submitting ipv4 allocator request: %s", err)
+		return subnetAllocator, err
+	}
+
+	log.Debugf("Response of ipv4 allocator %s", data)
+	if err := json.Unmarshal([]byte(data), &subnetAllocator); err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error with task un-marshal: %s", err)
+		return subnetAllocator, err
+	}
+
+	return subnetAllocator, nil
+}
+
+func (c *OVClient) CollectIpv4Subnet(id string, subnet SubnetCollectorList) (SubnetCollectorList, error) {
+	var (
+		uri             = "/rest/id-pools/ipv4/subnets/" + id + "/collector"
+		t               *Task
+		subnetCollector SubnetCollectorList
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t = t.NewProfileTask(c)
+	t.ResetTask()
```

Comment:
same as above, make sure youre utilizing task resource here, i dont see we are waiting till the task is completed.